tools/missing-test-detector/changed.go

```diff
@@ -66,8 +78,12 @@ func changedFieldsUnder(oldFieldSchema, newFieldSchema *schema.Schema) any {
 			return changedFields(nil, newFieldSchemaElem, true)
 		}
 	}
-	if oldFieldSchema == nil || newFieldSchema.Type != oldFieldSchema.Type || newFieldSchema.Elem != oldFieldSchema.Elem {
-		return false
+	if oldFieldSchema == nil {
+		return &Field{Added: true}
+	}
+	if oldFieldSchema.Type != newFieldSchema.Type {
+		return &Field{Changed: true}
 	}
+	// TODO(thomasrodgers) handle the case where something under Elem changed but Elem is not a schema.Resource
```

Comment:
Is this a blocker for this PR? What kind of case would fall under this?

Reply:
It shouldn't be a blocker here. The only case that needs to be handled is one where the type of elements in a list or map changes between versions, which should be pretty rare.

Reply:
👍  happy to merge without this fix but it would be great to fix in a follow-up PR.